Azure.ResourceManager.Core.Tests/LocationTests.cs

```diff
@@ -1,110 +1,173 @@
 ﻿using NUnit.Framework;
+using System;
 
 namespace Azure.ResourceManager.Core.Tests
 {
     public class LocationTests
     {
-        [Test]
-        public void CanCreateLocation()
-        {
-            Location tempLocation = new Location("West US");
-            Assert.AreEqual("West US", tempLocation.Name);
-        }
-
-        [Test]
-        public void EqualsToObjectTrue()
-        {
-            Location location1 = new Location("West US");
-            Location location2 = new Location("West US");
-
-            Assert.IsTrue(location1.Equals(location1));
-            Assert.IsTrue(location1.Equals(location2));
-        }
-
-        [Test]
-        public void EqualsToStringTrue()
-        {
-            Location location1 = new Location("West US");
-            Assert.IsTrue(location1.Equals("West US"));
-        }
-
-        [Test]
-        public void EqualsToObjectFalse()
-        {
-            Location location1 = new Location("West US");
-            Location location2 = new Location("South US");
-
-            Assert.IsFalse(location1.Equals(location2));
-        }
-
-        [Test]
-        public void EqualsToStringFalse()
+        [TestCase("westus", "westus", "west-us", "West US")]
+        [TestCase("west-us", "westus", "west-us", "West US")]
+        [TestCase("West US", "westus", "west-us", "West US")]
+        [TestCase("privatecloud", "privatecloud", "privatecloud", "privatecloud")]
+        [TestCase("private-cloud", "privatecloud", "private-cloud", "Private Cloud")]
+        [TestCase("Private Cloud", "privatecloud", "private-cloud", "Private Cloud")]
+        [TestCase("@$!()*&", "@$!()*&", "@$!()*&", "@$!()*&")]
+        [TestCase("W3$t U$ 2", "W3$t U$ 2", "W3$t U$ 2", "W3$t U$ 2")]
+        [TestCase("", "", "", "")]
+        [TestCase(" ", " ", " ", " ")]
+        [TestCase(null, null, null, null)]
+        public void CanCreateLocation(string name, string expectedName, string expectedCanonincalName, string expectedDisplayName)
         {
-            Location location1 = new Location("West US");
-            Assert.IsFalse(location1.Equals("WestUS"));
+            Location location = name;
+            if (name == null)
+            {
+                Assert.Throws<System.NullReferenceException>(() => { string x = location.Name; });
```

Comment:
this isn't needed because its not testing any lines of code from Location.cs

Instead just Assert.IsNull(location);